server/application/application.go

```diff
@@ -140,52 +141,98 @@ func NewServer(
 // getAppEnforceRBAC gets the Application with the given name in the given namespace. If no namespace is
 // specified, the Application is fetched from the default namespace (the one in which the API server is running).
 //
-// If the Application does not exist, then we have no way of determining if the user would have had access to get that
-// Application. Verifying access requires knowing the Application's name, namespace, and project. The user may specify,
-// at minimum, the Application name.
+// If the user does not provide a "project," then we have to be very careful how we respond. If an app with the given
+// name exists, and the user has access to that app in the app's project, we return the app. If the app exists but the
+// user does not have access, we return "permission denied." If the app does not exist, we return "permission denied" -
+// if we responded with a 404, then the user could infer that the app exists when they get "permission denied."
```

Comment:
nit: In the last statement, it needs to be `they get "permission denied."` -> `they should get "permission denied."`

Reply:
```suggestion
// if we responded with a 404, then the user could infer that the app exists when they get "permission denied" instead.
```

Hm. I think this might be closer to what I intended. wdyt?

Reply:
This works too..